### PR TITLE
feat: #896 sidenav collapseble for mobile

### PR DESCRIPTION
**Pull request by J0taFerreira**
- SideNav now collapse on mobile(390px) and tablet screens(768px).
- Refactored the sidnav to use primevue component

note: Some of sidebar primevue component styling will be on common repo

### CODE REVIEW

**Comment by ianliuwk1019** [comment]
Hi Jota, the PR shows this:
image
Not sure where the conflicts it means, maybe the branch is outdated or something else I am not sure what it is, please check.

**Comment by ianliuwk1019** [comment]
Hi Jota, I will check code later. But first when login, now it does not work for desktop browser (most users).
Or is it due to some common style (I don't think that's the case but I am not sure)?
This is what I see and it's very strange:
image

When I close the sidenav, it is like this on desktop browser:
image

Even if I use dev tool with responsive range, it is the same:
image

Please make sure again and try catching for major issues if you can or discuss if task is not clear before opening PR.

**Comment by J0taFerreira** [comment]
> Hi Jota, I will check code later. But first when login, now it does not work for desktop browser (most users). Or is it due to some common style (I don't think that's the case but I am not sure)? This is what I see and it's very strange: image
> 
> When I close the sidenav, it is like this on desktop browser: image
> 
> Even if I use dev tool with responsive range, it is the same: image
> 
> Please make sure again and try your best catching for major issues if you can or discuss if task is not clear before opening PR.

Hello, Ian! Some styling is on the common repo due because now the sideNav is using the sidebar comonent from primevue, I'll take a look on the problems you mentioned to see if it's not related to the common repo styling. Thanks for the comment!

**Comment by J0taFerreira** [comment]
> Hi Jota, I will check code later. But first when login, now it does not work for desktop browser (most users). Or is it due to some common style (I don't think that's the case but I am not sure)? This is what I see and it's very strange: image
> 
> When I close the sidenav, it is like this on desktop browser: image
> 
> Even if I use dev tool with responsive range, it is the same: image
> 
> Please make sure again and try your best catching for major issues if you can or discuss if task is not clear before opening PR.

I fixed the styling that got broken when updated the branch and seens to be working fine, some of the styling will still be broken because it's on the common repo.

**Comment by MCatherine1994** [comment]
This looks good to me with the new stylesheet in the common repo, we will need to release a new version over there. Only thing left is the page title, we want to show full application not just "FAM"
Screen Shot 2023-10-05 at 12 26 40 PM

**Comment by J0taFerreira** [comment]
> This looks good to me with the new stylesheet in the common repo, we will need to release a new version over there. Only thing left is the page title, we want to show full application not just "FAM" Screen Shot 2023-10-05 at 12 26 40 PM

Just made a commit with the fix, thanks you so much!

**Comment by MCatherine1994** [comment]
Hi Jota, let's pull from main, we always need to keep our branch up to date before merge

**Comment by ianliuwk1019** [comment]
> > Hi Jota, I will check code later. But first when login, now it does not work for desktop browser (most users). Or is it due to some common style (I don't think that's the case but I am not sure)? This is what I see and it's very strange: image
> > When I close the sidenav, it is like this on desktop browser: image
> > Even if I use dev tool with responsive range, it is the same: image
> > Please make sure again and try your best catching for major issues if you can or discuss if task is not clear before opening PR.
> 
> Hello, Ian! Some styling is on the common repo due because now the sideNav is using the sidebar comonent from primevue, I'll take a look on the problems you mentioned to see if it's not related to the common repo styling. Thanks for the comment!

Thanks Jota! Looks like once again is problem on my end without being able to get styling. But you show up in scrum, so all good.

**Review by ianliuwk1019 — CHANGES_REQUESTED** [review]

File: frontend/src/components/common/SideNav.vue [hunk 1]
```diff
@@ -23,67 +25,75 @@ const props = defineProps({
...
-                src="@/assets/images/17_gov3_bc_logo_transparent.svg"
-                alt="B.C. Government Logo"
-            />
-        </a>
-        <div class="content">
-            <ul>
-                <div v-for="item in props.data">
-                    <li class="header">{{ item.name }}</li>
+    <Sidebar
+        v-model:visible="useSideNavVisible.isSideNavVisible"
```
[the remark this answers is not in the document]
Comment (J0taFerreira) [reply frontend/src/components/common/SideNav.vue:32]: Some stateManagers such as pinia recommend naming the folder "store" and start the state variable name with "use[some_state]". But if is gonna cause trouble on other devs understanding maybe is better to change the name.

File: frontend/src/components/header/Header.vue [hunk 2]
```diff
@@ -67,6 +63,20 @@ const props = defineProps({
...
+li {
+    list-style: none;
+}
+
+.toggleSideNav {
+    margin-right: 0.5rem;
+    padding: 0.88rem;
+    border: none;
+    border-radius: 0;
+}
```
[the remark this answers is not in the document]
Comment (J0taFerreira) [reply frontend/src/components/header/Header.vue:76]: I can do that. Thanks!

File: frontend/src/components/managePermissions/ManagePermissions.vue [hunk 3]
```diff
@@ -95,7 +95,7 @@ async function deleteUserRoleAssignment(
                 @change="selectApplication"
                 :options="selectApplicationOptions"
                 optionLabel="application_description"
-                placeholder="Choose an application to manage permissions"
+                placeholder="Choose or enter an application to manage permissions"
```
[the remark this answers is not in the document]
Comment (J0taFerreira) [reply frontend/src/components/managePermissions/ManagePermissions.vue:98]: Sorry, I just copy from figma, I'll fix it. Thanks!

File: frontend/src/store/SideNavState.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { reactive } from 'vue'
+
+export const useSideNavVisible = reactive({
+    isSideNavVisible: false,
+    isDesktopSize: false,
+    toggleSideNavVisible() {
+        this.isSideNavVisible = !this.isSideNavVisible
+    },
+    isDesktop() {
+        return window.innerWidth >= 1024 && true
```
[the remark this answers is not in the document]
Comment (J0taFerreira) [reply frontend/src/store/SideNavState.ts:10]: This is some older code I copied, the "&& true" is not needed, nice catch!

File: frontend/src/components/managePermissions/UserDataTable.vue [hunk 5]
```diff
@@ -232,8 +232,15 @@ function deleteAssignment(assignment: FamApplicationUserRoleAssignmentGet) {
     border: none;
 }
 
+.data-table-container {
+    margin: 0.05rem;
+    background: $light-layer-one;
+    z-index: -1;
+    min-height: calc(100vh - 18.745rem);
```
Comment (MCatherine1994) [inline frontend/src/components/managePermissions/UserDataTable.vue:239]: Hi Jota, if you check our main branch, we don't set background color here anymore. The background color is set in the `frontend/src/components/managePermissions/ManagePermissions.vue` file. 
Screen Shot 2023-10-10 at 1 14 38 PM

Your changes break the style for notification message, currently on your branch, the notification message margin is not correct.
image

But if you check the FAM-dev, it looks good:
image

Please check if you really need this `data-table-container` class anymore.

Comment (J0taFerreira) [reply frontend/src/components/managePermissions/UserDataTable.vue:239]: Sure, I'll be looking into that to confirm if it's necessary. Thanks!

File: frontend/src/store/SideNavState.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+window.addEventListener("resize", (event) => {
+    screenSize = window.innerWidth
+    sideNavState.isVisible = sideNavState.isDesktop()
+    sideNavState.isDesktopSize = sideNavState.isDesktop()
+});
+
+export const sideNavState = shallowReactive({
+    isVisible: screenSize >= 1024 ? true : false,
+    isDesktopSize: screenSize >= 1024,
```
Comment (MCatherine1994) [inline frontend/src/store/SideNavState.ts:13]: um.... so I feel isVisible and isDesktopSize default should both depend on the screen size, let's code them in a consistent way. Either put "true : false" or both, or remove it.

Comment (ianliuwk1019) [reply frontend/src/store/SideNavState.ts:13]: Yes, I was looking at that and it is strange. Also, one state variable might be enough. And you don't need "true: false" on line 12, just comparing to line 13.

Comment (J0taFerreira) [reply frontend/src/store/SideNavState.ts:13]: sure, one state is indeed enough.

File: frontend/package.json [hunk 7]
```diff
@@ -13,13 +13,13 @@
...
     "dependencies": {
-        "@bcgov-nr/nr-theme": "^1.4.2",
+        "@bcgov-nr/nr-theme": "^1.5.0",
         "@carbon/icons-vue": "^10.72.1",
         "aws-amplify": "^5.3.6",
         "axios": "0.26.1",
         "bootstrap": "^5.2.3",
         "fam-api": "file:../client-code-gen/gen",
-        "primevue": "^3.27.0",
+        "primevue": "3.34.0",
```
Comment (MCatherine1994) [inline frontend/package.json:22]: The latest primevue version v3.36.0 has some style change that we need to adjust. For now we can stick with v3.34.0.

**Review by MCatherine1994 — APPROVED** [review]

**Review by ianliuwk1019 — APPROVED** [review]

**Pull request merged**